relayer/relays/beacon/header/syncer/api/api_response.go

```diff
@@ -515,6 +556,302 @@ func (b BeaconBlockResponse) ToScale() (scale.BeaconBlock, error) {
+	if err != nil {
+		return nil, err
+	}
+
+	proposerIndex, err := util.ToUint64(data.ProposerIndex)
+	if err != nil {
+		return nil, err
+	}
+
+	parentRoot, err := util.HexStringToByteArray(data.ParentRoot)
+	if err != nil {
+		return nil, err
+	}
+
+	stateRoot, err := util.HexStringToByteArray(data.StateRoot)
+	if err != nil {
+		return nil, err
+	}
+
+	body := data.Body
+
+	randaoReveal, err := util.HexStringToByteArray(body.RandaoReveal)
+	if err != nil {
+		return nil, err
+	}
+
+	eth1DepositRoot, err := util.HexStringToByteArray(body.Eth1Data.DepositRoot)
+	if err != nil {
+		return nil, err
+	}
+
+	eth1DepositCount, err := util.ToUint64(body.Eth1Data.DepositCount)
+	if err != nil {
+		return nil, err
+	}
+
+	eth1BlockHash, err := util.HexStringToByteArray(body.Eth1Data.BlockHash)
+	if err != nil {
+		return nil, err
+	}
+
+	graffiti, err := util.HexStringTo32Bytes(body.Graffiti)
+	if err != nil {
+		return nil, err
+	}
+
+	proposerSlashings := []*state.ProposerSlashing{}
+	for _, proposerSlashing := range body.ProposerSlashings {
+		proposerSlashingSSZ, err := proposerSlashing.ToFastSSZ()
+		if err != nil {
+			return nil, err
+		}
+
+		proposerSlashings = append(proposerSlashings, proposerSlashingSSZ)
+	}
+
+	attesterSlashings := []*state.AttesterSlashing{}
+	for _, attesterSlashing := range body.AttesterSlashings {
+		attesterSlashingSSZ, err := attesterSlashing.ToFastSSZ()
+		if err != nil {
+			return nil, err
+		}
+
+		attesterSlashings = append(attesterSlashings, attesterSlashingSSZ)
+	}
+
+	attestations := []*state.Attestation{}
+	for _, attestation := range body.Attestations {
+		attestationSSZ, err := attestation.ToFastSSZ()
+		if err != nil {
+			return nil, err
+		}
+
+		attestations = append(attestations, attestationSSZ)
+	}
+
+	deposits := []*state.Deposit{}
+	for _, deposit := range body.Deposits {
+		depositScale, err := deposit.ToFastSSZ()
+		if err != nil {
+			return nil, err
+		}
+
+		deposits = append(deposits, depositScale)
+	}
+
+	voluntaryExits := []*state.SignedVoluntaryExit{}
+	for _, voluntaryExit := range body.VoluntaryExits {
+		voluntaryExitSSZ, err := voluntaryExit.ToFastSSZ()
+		if err != nil {
+			return nil, err
+		}
+
+		voluntaryExits = append(voluntaryExits, voluntaryExitSSZ)
+	}
+
+	executionPayload := body.ExecutionPayload
+	parentHash, err := util.HexStringTo32Bytes(executionPayload.ParentHash)
+	if err != nil {
+		return nil, err
+	}
+
+	executionStateRoot, err := util.HexStringTo32Bytes(executionPayload.StateRoot)
+	if err != nil {
+		return nil, err
+	}
+
+	receiptsRoot, err := util.HexStringTo32Bytes(executionPayload.ReceiptsRoot)
+	if err != nil {
+		return nil, err
+	}
+
+	prevRando, err := util.HexStringTo32Bytes(executionPayload.PrevRandao)
+	if err != nil {
+		return nil, err
+	}
+
+	n := new(big.Int)
+	n, ok := n.SetString(executionPayload.BaseFeePerGas, 10)
+	if !ok {
+		return nil, err
+	}
+
+	// FastSSZ expects a little endian byte array
+	baseFeePerGas := util.ChangeByteOrder(n.Bytes())
```

Comment:
Interesting, Is this  because `baseFeePerGas` is actually a `uint256` in spec?

Reply:
Yes, exactly! 😄 And in the spec unsigned ints are encoded to a 32 little endian byte, and big Int seems to use big endian. A bit of a pain, but we can remove this when we can convert the Lodestar block ssz bytes directly to the FastSSZ structs.